.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+---
+name: Shellcheck
+
+on:
+  push:
+    branches: [develop]
+  pull_request:
+    branches: [develop]
+
+permissions:
+  contents: read
+
+jobs:
+  shellcheck:
+    name: Check shell scripts
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          sudo apt update && sudo apt install -y shellcheck
+      - name: shellcheck
+        run: |
+          git grep -l '^#\( *shellcheck \|!\(/bin/\|/usr/bin/env \)\(sh\|bash\|dash\|ksh\)\)' | xargs shellcheck
```

Comment:
Love this!

Just to tidy up, do you think it makes sense to move this into the [`.github/workflows/lint.yml`](https://github.com/mermaid-js/mermaid/blob/051ef598f669a40b783e3566631d7b31f8f1ab9f/.github/workflows/lint.yml) with the other lint job?

You'll have to move the `permissions: contents: read` into the `jobs.shellcheck` too.